emperor/support_files/js/view.js

```diff
@@ -353,6 +378,130 @@ DecompositionView.prototype._fastInit = function() {
   this.markers.push(cloud);
 };
 
+/**
+ * Parallel plots closely mirroring the shader enabled _fastInit calls
+ */
+DecompositionView.prototype._fastInitParallelPlot = function()
+{
+  var positions, colors, opacities, visibilities, geometry, cloud;
+
+  // We're really just drawing a bunch of line strips...
+  // highly doubt shaders are necessary for this...
+  var vertexShader = `
+    attribute vec3 color;
+    attribute float opacity;
+    attribute float visible;
+    
+    varying vec3 vColor;
+    varying float vOpacity;
+    varying float vVisible;
+    
+    void main() {
+      vColor = color;
+      vOpacity = opacity;
+      vVisible = visible;
+    
+      gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.0);
+    }
+    `;
+
+  var fragmentShader = `
+    precision mediump float;
+    varying vec3 vColor;
+    varying float vOpacity;
+    varying float vVisible;
+    
+    void main() {
+      if (vVisible <= 0.0 || vOpacity <= 0.0)
+        discard;
+      gl_FragColor = vec4(vColor, vOpacity);
+    }
+  `;
+
+  var allDimensions = _.range(this.decomp.dimensions);
+
+  // We'll build the line strips as GL_LINES for simplicity, at least for now,
+  // by doubling up vertex positions at each of the intermediate axes.
+  var numPoints = (allDimensions.length * 2 - 2) * (this.decomp.length);
+  positions = new Float32Array(numPoints * 3);
+  colors = new Float32Array(numPoints * 3);
+  opacities = new Float32Array(numPoints);
+  visibilities = new Float32Array(numPoints);
+
+  var material = new THREE.ShaderMaterial({
+    vertexShader: vertexShader,
+    fragmentShader: fragmentShader,
+    transparent: true
+  });
+
+  geometry = new THREE.BufferGeometry();
+  geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
+  geometry.addAttribute('color', new THREE.BufferAttribute(colors, 3));
+  geometry.addAttribute('opacity', new THREE.BufferAttribute(opacities, 1));
+  geometry.addAttribute('visible', new THREE.BufferAttribute(visibilities, 1));
+
+  lines = new THREE.LineSegments(geometry, material);
+
+  var attributeIndex = 0;
+
+  for (var i = 0; i < this.decomp.length; i++)
+  {
+    var plottable = this.decomp.plottable[i];
+    // Each point in the model maps to (allDimensions.length * 2 - 2)
+    // positions due to the use of lines rather than line strips.
+    for (var j = 0; j < allDimensions.length; j++)
+    {
+      //normalize by global range bounds
+      var globalMin = this.allModels.dimensionRanges.min[allDimensions[j]];
+      var globalMax = this.allModels.dimensionRanges.max[allDimensions[j]];
+      var maxMinusMin = globalMax - globalMin;
+      var interpVal = (plottable.coordinates[j] - globalMin) / (maxMinusMin);
+      geometry.attributes.position.setXYZ(attributeIndex,
+                                        j,
```

Comment:
Any chance you could align the indentation of these arguments? And the ones in line 473?